### ISSUE TITLE
Unable to run @SpringBootTest without WebMVC

### ISSUE THREAD

**Issue by C-Otto** [springfox/springfox#1935]
Running a very simple Spring Boot Test fails due to Springfox:

```
@RunWith(SpringRunner.class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE)
public class Foo {
    @Test
    public void test() throws Exception {
    }
}
```

The reason for this is related to #1611 as no WebMVC beans are available. However, these are required for Springfox.

Version 2.6.1

Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'documentationPluginsBootstrapper' defined in URL [jar:file:/C:/LocalData/dev/repo/io/springfox/springfox-spring-web/2.6.1/springfox-spring-web-2.6.1.jar!/springfox/documentation/spring/web/plugins/DocumentationPluginsBootstrapper.class]: Unsatisfied dependency expressed through constructor parameter 1;

nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'webMvcRequestHandlerProvider' defined in URL [jar:file:/C:/LocalData/dev/repo/io/springfox/springfox-spring-web/2.6.1/springfox-spring-web-2.6.1.jar!/springfox/documentation/spring/web/plugins/WebMvcRequestHandlerProvider.class]: Unsatisfied dependency expressed through constructor parameter 0;

nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'java.util.List<org.springframework.web.servlet.mvc.method.RequestMappingInfoHandlerMapping>' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}

**Comment by dilipkrish** [comment]
Can you exclude the `@EnableSwagger2` annotation that is pulled in automatically into this test?

**Comment by C-Otto** [comment]
What do you mean with "exclude", is there a Spring property for that? I did not include any Springfox annotation in my own code, but I am using a Maven parent project that pulls it in (and most likely is the reason for its activation).

**Comment by dilipkrish** [comment]
When you do a `@SpringBootTest` it includes your application configuration. You have to tell it to exclude specific configuration or give it a test configuration. The idea is to not include the swagger configuration in the test configuration.

**Comment by C-Otto** [comment]
OK, I understand what you mean. Explicitly excluding configurations does not work, as far as I know. Besides, having to do this in every test class seems wrong to me. Would it be possible to only start Springfox in a MVC environment, using conditional bean configuration (e.g. `@ConditionalOnClass`)? That's the way Spring Boot works under the hood.

**Comment by dilipkrish** [comment]
@C-Otto couple of things to try. Create a configuration that you explicitly include for your tests that you can share across your tests. Also keep in mind that this is a spring library that is also used in spring boot applications. So the idea of bringing in a dependency on spring boot will not work. It has to be something in the client side testing that needs to handle this.

**Comment by C-Otto** [comment]
OK, I see. Under these circumstances I'd say my bug report is invalid. Thanks for the help!

**Comment by kilaka** [comment]
Why is springfox behaves badly with junit?

**Comment by markandirene** [comment]
How do I exclude @EnableSwagger2 from a test

**Comment by dilipkrish** [comment]
This has not been fixed #2329

**Comment by kalyanraavi** [comment]
@dilipkrish 
hi dilip,
I am facing the same issue even though I am using 2.9.2 version of Springfox.  Here is the error. can you please advise.

Expecting:
 <Unstarted application context org.springframework.boot.test.context.assertj.AssertableWebApplicationContext[startupFailure=org.springframework.beans.factory.UnsatisfiedDependencyException]>
to have bean named:
 <"customDocketProvider">:
but context failed to start:
 org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'documentationPluginsBootstrapper' defined in URL [jar:file:/Users//.gradle/caches/modules-2/files-2.1/io.springfox/springfox-spring-web/2.9.2/ed2ed714a6cba8804d00f80f0534901e4c7a3211/springfox-spring-web-2.9.2.jar!/springfox/documentation/spring/web/plugins/DocumentationPluginsBootstrapper.class]: Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'webMvcRequestHandlerProvider' defined in URL [jar:file:/Users/.gradle/caches/modules-2/files-2.1/io.springfox/springfox-spring-web/2.9.2/ed2ed714a6cba8804d00f80f0534901e4c7a3211/springfox-spring-web-2.9.2.jar!/springfox/documentation/spring/web/plugins/WebMvcRequestHandlerProvider.class]: Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'java.util.List<org.springframework.web.servlet.mvc.method.RequestMappingInfoHandlerMapping>' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
